crates/ruma-common/src/push/condition.rs

```diff
@@ -137,6 +141,16 @@ pub enum PushCondition {
         value: ScalarJsonValue,
     },
 
+    /// Matches a thread event based on the user's thread subscription status, as defined by
+    /// [MSC4306].
+    ///
+    /// [MSC4306]: https://github.com/matrix-org/matrix-spec-proposals/pull/4306
+    #[cfg(feature = "unstable-msc4306")]
+    ThreadSubscription {
+        /// Whether the user must be subscribed to the thread for the condition to match.
```

Comment:
Maybe we could clarify this to avoid Jonas' confusion:

```suggestion
        /// Whether the user must be subscribed (`true`) or unsubscribed (`false`) to the thread for the condition to match.
```